executor/aggfuncs/func_max_min.go

```diff
@@ -299,22 +429,26 @@ func (e *maxMin4Int) MergePartialResult(sctx sessionctx.Context, src, dst Partia
 
 type maxMin4IntSliding struct {
 	maxMin4Int
+	windowInfo
 }
 
 func (e *maxMin4IntSliding) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
 	p := (*partialResult4MaxMinInt)(pr)
-	for _, row := range rowsInGroup {
+	for i, row := range rowsInGroup {
 		input, isNull, err := e.args[0].EvalInt(sctx, row)
 		if err != nil {
 			return 0, err
 		}
 		if isNull {
 			continue
 		}
-		p.heap.Append(input)
+		err = p.deque.Enqueue(uint64(i)+e.start, input)
```

Comment:
Deque needs the absolute position of each element, here `i` only denotes the relative position in `rowsInGroup`. To get the absolute position, we need to add offset of `e.start`, which represents the absolute index of start of window.